### ISSUE TITLE
Let’s Encrypt Cert Expiring Soon (Nov. 23)

### ISSUE THREAD

**Issue by captbaritone** [vim-awesome/vim-awesome#398]
Maybe the server is not correctly renewing certs?

Just got this email: 

---
Your certificate (or certificates) for the names listed below will expire in 10 days (on 23 Nov 21 23:21 +0000). Please make sure to renew your certificate before then, or visitors to your web site will encounter errors.

We recommend renewing certificates automatically when they have a third of their total lifetime left. For Let's Encrypt's current 90-day certificates, that means renewing 30 days before expiration. See https://letsencrypt.org/docs/integration-guide/ for details.

vimawesome.com

For any questions or support, please visit: https://community.letsencrypt.org/ Unfortunately, we can't provide support by email.

For details about when we send these emails, please visit: https://letsencrypt.org/docs/expiration-emails/ In particular, note that this reminder email is still sent if you've obtained a slightly different certificate by adding or removing names. If you've replaced this certificate with a newer one that covers more or fewer names than the list above, you may be able to ignore this message.

**Comment by florianb** [comment]
And its finally outdated – do you need a helping hand fixing it?

**Comment by captbaritone** [comment]
I tried logging in and this is the error message I see:

```
Traceback (most recent call last):
  File "/usr/bin/certbot", line 11, in <module>
    load_entry_point('certbot==0.31.0', 'console_scripts', 'certbot')()
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 1365, in main
    return config.func(config, plugins)
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 1111, in run
    le_client = _init_le_client(config, authenticator, installer)
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 612, in _init_le_client
    return client.Client(config, acc, authenticator, installer, acme=acme)
  File "/usr/lib/python3/dist-packages/certbot/client.py", line 266, in __init__
    acme = acme_from_config_key(config, self.account.key, self.account.regr)
  File "/usr/lib/python3/dist-packages/certbot/client.py", line 51, in acme_from_config_key
    return acme_client.BackwardsCompatibleClientV2(net, key, config.server)
  File "/usr/lib/python3/dist-packages/acme/client.py", line 814, in __init__
    directory = messages.Directory.from_json(net.get(server).json())
  File "/usr/lib/python3/dist-packages/acme/client.py", line 1152, in get
    self._send_request('GET', url, **kwargs), content_type=content_type)
  File "/usr/lib/python3/dist-packages/acme/client.py", line 1101, in _send_request
    response = self.session.request(method, url, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 502, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 612, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: ("bad handshake: Error([('SSL routines', 'SSL3_GET_SERVER_CERTIFICATE', 'certificate verify failed')],)",)
2021-11-26 18:03:32,702:ERROR:certbot.log:An unexpected error occurred:
```

I figured we'd need to upgrade certbot, but it looks like certbot was installed using pip (maybe by me years ago?) and that's not the recommended way to do it. I tried looking into installing certbot the suggested way via snapd, but I think our version of Ubuntu is too old to do it without some special care that I'm not sure about.

I looked into just upgrading via pip `pip install --upgrade certbot` and got:

```
[...]
Downloading/unpacking cryptography>=2.1.4 (from certbot)
  Downloading cryptography-36.0.0.tar.gz (571kB): 571kB downloaded
  Running setup.py (path:/tmp/pip_build_vim/cryptography/setup.py) egg_info for package cryptography
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/tmp/pip_build_vim/cryptography/setup.py", line 84
        print(f"    Python: {'.'.join(str(v) for v in sys.version_info[:3])}")
                                                                            ^
    SyntaxError: invalid syntax
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/tmp/pip_build_vim/cryptography/setup.py", line 84

    print(f"    Python: {'.'.join(str(v) for v in sys.version_info[:3])}")

                                                                        ^

SyntaxError: invalid syntax

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /tmp/pip_build_vim/cryptography
Storing debug log for failure in /tmp/tmpSEQ6pQ
```

I suspect this is something to do with trying to run Python 3 code using Python 3?

I don't have any more time to look into this today, so I'll have to leave it here.

cc @sophiebits @divad12

**Comment by joaquinco** [comment]
@captbaritone Maybe this is the problem: https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/

**Comment by sophiebits** [comment]
Fixed, for now.

* ran `sudo python -m pip install josepy==1.1.0 requests==2.14.2 acme==0.29.0 certbot==0.32.0 certbot-nginx==0.32.0` to get the old packages back
* manually edited `/usr/local/lib/python2.7/dist-packages/certifi/cacert.pem` to trust ISRG Root X1, which is Let's Encrypt's current root CA (I also added `/usr/local/share/ca-certificates/isrgrootx1.crt` and ran `update-ca-certificates` before figuring out that certifi's store is the one I needed to update)
* ran `certbot certonly --nginx -d vimawesome.com --non-interactive`, the command in the crontab
* `/usr/init.d/nginx reload`

Works now for me.

We gotta upgrade the machine and/or move to something easier to maintain. (Maybe it's worth sitting down to rip out RethinkDB and then using a common serverless platform?)

**Comment by Kavinjsir** [comment]
Hey, not sure if this is relative.
Just seems like an outdated cert when viewing https://vimawesome.com/
<img width="382" alt="Screen Shot 2022-02-27 at 10 58 16 AM" src="https://user-images.githubusercontent.com/18136486/155889762-192d53ed-c62a-46bd-b246-d5f0a030b9e4.png">
Would be glad if someone may help.

**Comment by lonepeon** [comment]
Hello.

Same as @Kavinjsir, I’m not sure it’s related but when we open vimawesome in Safari we get this error:

> Safari warns you when a website has
an expired certificate. This website's
certificate expired 5 days ago.


This has just been reported in [r/vim](https://www.reddit.com/r/vim/comments/t649ds/vimawesome_site_accessing_issue/).

**Comment by l0rb** [comment]
@sophiebits seems to have fixed this temporarily. The certificate currently used expired on the 26. of February and thus most browser give users a warning or outright block the site.

**Comment by AlienResidents** [comment]
> @sophiebits seems to have fixed this temporarily.

How so?  The TLS cert is still expired for me.

**Comment by l0rb** [comment]
> > @sophiebits seems to have fixed this temporarily.
> 
> How so? The TLS cert is still expired for me.

The issue first occured on 23.11.2021 than was temporarily fixed from 29.11.2021 until 26.02.2022, and since than it's broken again.
